server/app/services/ti/TrustedIntermediaryService.java

```diff
@@ -0,0 +1,100 @@
+ * form has no-errors, it sends back a successful TiclientCreationResult object.
+ *
+ * <p>If any of the validation fails, it sends a failed TIClientCreationResult object with the form
+ * and all of its errors.
+ */
+public final class TrustedIntermediaryService {
+  private final UserRepository userRepository;
+  private final DateConverter dateConverter;
+
+  @Inject
+  public TrustedIntermediaryService(UserRepository userRepository, DateConverter dateConverter) {
+    this.userRepository = Preconditions.checkNotNull(userRepository);
+    this.dateConverter = Preconditions.checkNotNull(dateConverter);
+  }
+
+  public TIClientCreationResult addNewClient(
+      Form<AddApplicantToTrustedIntermediaryGroupForm> form,
+      TrustedIntermediaryGroup trustedIntermediaryGroup) {
+    form = validateEmailAddress(form);
+    form = validateFirstName(form);
+    form = validateLastName(form);
+    form = validateDateOfBirth(form);
+    if (form.hasErrors()) {
+      return TIClientCreationResult.failure(form);
+    }
+    try {
+      userRepository.createNewApplicantForTrustedIntermediaryGroup(
+          form.get(), trustedIntermediaryGroup);
+    } catch (EmailAddressExistsException e) {
+      return TIClientCreationResult.failure(
+          form.withError(
+              "emailAddress",
+              "Email address already in use. Cannot create applicant if an account already"
+                  + " exists."));
+    }
+    return TIClientCreationResult.success();
+  }
+
+  private Form<AddApplicantToTrustedIntermediaryGroupForm> validateEmailAddress(
+      Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getEmailAddress())) {
+      return form.withError("emailAddress", "Email Address required");
+    }
+    return form;
+  }
+
+  private Form<AddApplicantToTrustedIntermediaryGroupForm> validateFirstName(
+      Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getFirstName())) {
+      return form.withError("firstName", "First name required");
+    }
+    return form;
+  }
+
+  private Form<AddApplicantToTrustedIntermediaryGroupForm> validateLastName(
+      Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getLastName())) {
+      return form.withError("lastName", "Last name required");
+    }
+    return form;
+  }
+
+  private Form<AddApplicantToTrustedIntermediaryGroupForm> validateDateOfBirth(
+      Form<AddApplicantToTrustedIntermediaryGroupForm> form) {
+    if (Strings.isNullOrEmpty(form.value().get().getDob())) {
+      return form.withError("dob", "Date of Birth required");
+    }
+    LocalDate currentDob = null;
+    try {
+      currentDob = dateConverter.parseStringToLocalDate(form.value().get().getDob());
+    } catch (DateTimeParseException e) {
+      return form.withError("dob", "Date of Birth must be in MM-dd-yyyy format");
```

Comment:
issue (blocking): is this validation message correct? i'm seeing the date in a different format everywhere else in this code

Reply:
So the date is entered as "MM-dd-yyyy" format through the date picker. But our J2HTML date picker converts it from mm-dd-yyyy format to yyyy-mm-dd format. check how we enter date:
![image](https://user-images.githubusercontent.com/101214724/185254371-0e84be48-b988-4bf5-9642-0c12a0f9d5b4.png)